### PR TITLE
dtls.c: fix processing of alerts.

### PR DESCRIPTION
**Pull request by boaks**
If alerts remove the peer, additional records need to be dropped.

### CODE REVIEW

File: dtls.c [hunk 1]
```diff
@@ -4267,6 +4267,7 @@ dtls_handle_message(dtls_context_t *ctx,
   }
 
   while ((rlen = is_record(msg,msglen))) {
+
```
Comment (obgm) [inline dtls.c:4270]: Why an empty line? I think this should be removed before merge.

File: dtls.c [hunk 2]
```diff
@@ -4395,8 +4407,8 @@ dtls_handle_message(dtls_context_t *ctx,
       } else {
         dtls_stop_retransmission(ctx, peer);
       }
-      break;
-
+      /* no more valid records after alerts */
+      return 0;
```
Comment (obgm) [inline dtls.c:4411]: There is [one case](https://github.com/eclipse/tinydtls/pull/190/commits/62c8c08850df36da7201871eb1e10b2d7c58268b#diff-947f252e5a6242133f45e2963433822253886eb9a6bf4c12c7163ac8e90fe017L4396) where `peer` is currently not set to `NULL` (which is correct, I think). Changing `break` to `return 0` here should not break things, but I wanted to mention this case in order to do a double-check.

File: dtls.c [hunk 3]
```diff
@@ -4282,6 +4283,17 @@ dtls_handle_message(dtls_context_t *ctx,
                  content_type, epoch, pkt_seq_nr, rlen);
     }
 
+    if (!peer) {
```
Comment (obgm) [inline dtls.c:4286]: Since re-validation is triggered always at the end of this while-loop, we could decide to move the initial retrieval into this loop.

**Review by obgm — APPROVED** [review]
LGTM, just a couple of minor comments to take into consideration.

File: dtls.c (hunk 2 above)
Comment (boaks) [reply dtls.c:4411]: We can keep the old function for DTLS_ALERT_LEVEL_WARNING. But that would only make a difference, if such a Alert/Warning and additional records are sent within the same UDP message.
I will change it.

File: dtls.c (hunk 1 above)
Comment (boaks) [reply dtls.c:4270]: OK.

File: dtls.c (hunk 3 above)
Comment (boaks) [reply dtls.c:4286]: I kept the copy for the different logging message. 
That could be also achieved, when the snippets are merged.
I will try it.

**Comment by boaks** [comment]
The comments are addressed.

File: dtls.c (hunk 2 above)
Comment (obgm) [reply dtls.c]: Correct, and I think that a message that contains additional records after the alert can be treated as broken—why would an application do that?

**Comment by obgm** [comment]
Still looks good :-)
I think we are ready to merge.

**Comment by boaks** [comment]
> I think we are ready to merge.

Yeap, ones the test builds have succeeded :-).

**Pull request merged**